lib/avromatic/model/types/union_type.rb

```diff
@@ -76,9 +76,8 @@ def referenced_model_classes
         private
 
         def find_index(value)
-          # TODO: Cache this?
```

Comment:
This can be a hot code path, so it would be nice to be able to cache this index lookup (i.e., build a cache from `value.class` to `union_index`). With this change, since we match on the `value` itself, that cache isn't possible.

Additionally, the `matched?` call adds overhead for all types, but we only need it for the case where `value.is_a?(Integer)`. What do you think about special casing integer handling here?

```ruby
def find_index(value)
  if value.is_a?(::Integer)
    if Avromatic::Model::Types::IntegerType.in_range?(value)
      member_types.index(Avromatic::Model::Types::IntegerType)
    end || if Avromatic::Model::Types::BigIntType.in_range?(value)
      member_types.index(Avromatic::Model::Types::BigIntType)
    end
  else
    # TODO: Cache this
    member_types.find_index do |member_type|
      member_type.value_classes.any? { |value_class| value.is_a?(value_class) }
    end
  end
end
```

Reply:
@erikkessler1 fair enough. I've added special casing for Integer.

Reply:
P.S. is there anything different for this CI run?

The same specs locally are running with no issues:

```
❯ bundle exec rspec --format progress spec
...............................................................................................................................................................................................................................................................................................................................................

Finished in 0.53541 seconds (files took 0.60385 seconds to load)
335 examples, 0 failures

Coverage report generated for RSpec to avromatic/coverage.
Line Coverage: 96.59% (1245 / 1289)
```

Reply:
It seems like the 403 is coming from Sinatra ([documentation](https://sinatrarb.com/intro.html#:~:text=if%20not%20available.-,host_authorization,-You%20can%20pass)) as the host doesn't match the `permitted_hosts`. This must have changed between Sinatra versions. I was able to get things passing locally by changing the `Avromatic.registry_url` [here](https://github.com/salsify/avromatic/blob/5899c5e7efdd7d2dec6b5b63951c2ea3cf831de4/spec/spec_helper.rb#L22):

```diff
-Avromatic.registry_url = 'http://username:password@registry.example.com'
+Avromatic.registry_url = 'http://username:password@registry.localhost'
```

Reply:
Updated.